python/ray/tune/trial.py

```diff
@@ -312,10 +312,25 @@ def should_checkpoint(self):
             return False
 
     def has_checkpoint(self):
-        return self._checkpoint.value is not None
+        return self.checkpoint.value is not None
+
+    def update_checkpoint(self, checkpoint_path):
```

Comment:
do we need this logic here?

Reply:
We need to keep track of the best checkpoint for recovery, this is just the previous `compare_checkpoints` + the checkpoint-setting code previously in `ray_trial_executor::save` written as a single function.